### PR TITLE
feat: Teleport DOT between Statemint and Polkadot

### LINKED ISSUE #6540: Teleport DOT to Statemint

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Feature

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #6540
  - Teleport DOT from Statemint to Polkdot
  tx: https://assethub-polkadot.subscan.io/xcm_message/polkadot-4b577bee7cc388bc291b5210472061f89131a33b
  - Teleport DOT from Polkdot to Statemint
  tx: https://assethub-polkadot.subscan.io/xcm_message/polkadot-f34f6cc60aaff122fd6a79aa65753bc4b8e84d3c

Please note that the Existential Deposit on asset hub is 0.1 dot

> Existential Deposit: The amount deposited to the recipient will be net the calculated cross-chain fee. If the recipient address is new, the amount deposited should be greater than the Existential Deposit
> ref: https://wiki.polkadot.network/docs/build-protocol-info#existential-deposit 

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

image

  ## Copilot Summary
  
### <samp>🤖 Generated by Copilot at b16850c</samp>

This pull request adds support for multiple origin chains for the teleport feature, allowing users to transfer NFTs between Kusama, Polkadot, and Statemint. It refactors the `Teleport.vue` component and updates the `useChain.ts` and `teleport.ts` utils to handle the different chains and currencies. It also modifies the `teleport.receiveValue` translation string to dynamically display the currency symbol.

  
### <samp>🤖 Generated by Copilot at b16850c</samp>

> _We're sailing across the polkadot seas_
> _With `teleport` we can move our money with ease_
> _We've got `KSM`, `DOT`, and `STATEMINT` too_
> _So pull on the rope and sing out the value_

### CODE REVIEW

**Comment by prury** [comment]
Wrong decimals on balance: 
image
image

i would rename Dot here because we are referring to the chain and not the token:
image

Ahp to Polkadot: 
https://assethub-polkadot.subscan.io/tx/0xb570eae0cf0bc384b944d12f34ca37c49eac7ee0e7cd1eb723cc9f017b529e16?tab=xcm_transfer ✔️ 

Polkadot to Ahp -> https://polkadot.subscan.io/tx/0x8705da52c61178aa20532c4053d443011c977821a5a86dca8ec9237976ba75c0 ✔️ 

@Jarsen136 lmk if you want to mess with ED, and I'll open an issue. (related: #7213)

**Comment by Jarsen136** [comment]
> Wrong decimals on balance: image image
> 

✅ FIXED

> i would rename Dot here because we are referring to the chain and not the token: image
> 

I would wait for #7263 to merge first because they are related, and then I will update its name. 

> 
> @Jarsen136 lmk if you want to mess with ED, and I'll open an issue. (related: #7213)

It's the same issue as #7213, so it could be solved there instead of creating a new issue.

**Comment by prury** [comment]
> > Wrong decimals on balance: image image
> 
> ✅ FIXED
> 
> > i would rename Dot here because we are referring to the chain and not the token: image
> 
> I would wait for #7263 to merge first because they are related, and then I will update its name.
> 
> > @Jarsen136 lmk if you want to mess with ED, and I'll open an issue. (related: #7213)
> 
> It's the same issue as #7213, so it could be solved there instead of creating a new issue.

okie, ty!

**Comment by Jarsen136** [comment]
> 
> > i would rename Dot here because we are referring to the chain and not the token: image
> 
> I would wait for #7263 to merge first because they are related, and then I will update its name.
> 

I changed my mind. The naming for Polkdot could be updated here. Thank you for pointing out. @prury 

image

File: components/teleport/Teleport.vue [hunk 1]
```diff
@@ -177,11 +211,32 @@ const toTabs = [
...
+  },
 ]
 
-const ksmTokenDecimals = computed(() => assets(5).decimals)
+const currentTokenDecimals = computed(() => {
+  switch (fromChain.value) {
+    case Chain.KUSAMA:
+    case Chain.BASILISK:
+    case Chain.STATEMINE:
+      return assets(5).decimals
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:231]: `const { tokenId } = usePrefix()` instead of `5`?

File: components/teleport/Teleport.vue [hunk 2]
```diff
@@ -117,22 +116,45 @@ const { assets } = usePrefix()
...
 const amount = ref() //Required amount to be transfered is stored here
-const currency = ref('KSM') //Selected currency is stored here
 const isLoading = ref(false)
 const unsubscribeKusamaBalance = ref()
 const resetStatus = () => {
   amount.value = undefined
   isLoading.value = false
 }
 
+const currency = computed(() => {
```
Comment (roiLeo) [inline components/teleport/Teleport.vue:130]: `chainSymbol` or `symbol` does not produce the expected result?

**Review by roiLeo — APPROVED** [review]
otherwise code lgtm

File: components/teleport/Teleport.vue (hunk 1 above)
Comment (Jarsen136) [reply components/teleport/Teleport.vue:231]: The tokenId from `usePrefix ` is based on the prefix on the url. It's not the same chain as what the user selected on teleport page.

File: components/teleport/Teleport.vue (hunk 2 above)
Comment (Jarsen136) [reply components/teleport/Teleport.vue:130]: No, it doesn't. The result from `useChain` is based on the prefix on the url. It's not what we need on teleport page.

**Comment by exezbcz** [comment]
I saw this in one of your pics
image


Otherwise, I think it is time to update teleport as well; there are too many chains for this v1. :D

**Comment by Jarsen136** [comment]
> I saw this in one of your pics image
> 

✅ Fixed. It's a compatibility issue that only happens on Firefox.

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.04 USD/DOT ~ 7.426 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x72f07ad7d422b05ee0341eef5e86dbef85ac8de6a30dbf512932b0ff096e6ea1](https://polkadot.subscan.io/extrinsic/0x72f07ad7d422b05ee0341eef5e86dbef85ac8de6a30dbf512932b0ff096e6ea1)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**